server/src/main/java/org/elasticsearch/TransportVersion.java

```diff
@@ -118,6 +118,13 @@ public static List<TransportVersion> getAllVersions() {
         return VersionsHolder.ALL_VERSIONS;
     }
 
+    /**
+     * @return whether the given {@code id} corresponds to a known {@link TransportVersion}.
+     */
+    public static boolean isKnownVersionId(int id) {
```

Comment:
Instead of passing around bare ids, could this be an instance method on TransportVersion?

Reply:
Good point, I thought I'd be doing this before I'd created the instance but turns out that was wrong.